sdk/Trace/TextMapGetterSetter.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Sdk\Trace;
+
+use function array_keys;
+use ArrayAccess;
+use function get_class;
+use function gettype;
+use InvalidArgumentException;
+use function is_array;
+use function is_object;
+use function is_string;
+use OpenTelemetry\Trace as API;
+use function sprintf;
+use function strtolower;
+
+/**
+ * @see https://github.com/open-telemetry/opentelemetry-specification/blob/v1.6.1/specification/context/api-propagators.md#textmap-propagator Getter and Setter.
+ *
+ * Default implementation of {@see API\PropagationGetter} and {@see API\PropagationSetter}.
+ * This type is used if no custom getter/setter is provided to {@see API\TextMapPropagator::inject()} or {@see API\TextMapPropagator::extract()}.
+ */
+class TextMapGetterSetter implements API\PropagationGetter, API\PropagationSetter
+{
+    /** @var self|null */
+    private static $instance;
+
+    /**
+     * Returns a singleton instance of `self` to avoid, multiple runtime allocations.
+     */
+    public static function getInstance(): self
+    {
+        if (null === self::$instance) {
+            self::$instance = new self();
+        }
+
+        return self::$instance;
+    }
+
+    /** {@inheritdoc} */
+    public function keys($carrier): array
+    {
+        if (is_array($carrier)) {
```

Comment:
This method is required by the API but was missing an implementation. Opted to only support arrays even tho the actual getter supports `ArrayAccess` as that interface doesn't expose a way to get all the keys so oh well.

Reply:
Yup; I like that plan.

Reply:
Option 2: Define another interface that extends `ArrayAccess` to expose the `keys`, then only accept that or array. This way we can still call it `ArrayGetterSetter` but support both built in arrays and array-like objects. Thoughts?

Reply:
What about `ArrayAccessGetterSetter` then it's clear it could be used for anything that can be accessed as an array?

Reply:
Sounds like a plan!